Driverless AI/scoring-pipeline-deployment-in-python-runtime/scoring-pipeline-deployment-in-python-runtime.md

```diff
@@ -552,23 +550,27 @@ Execute the python script, so the TCP client local to the TCP server will get pr
 # Activate virtual environment env to run python example.py
 source tcp_client_env/bin/activate
 python custom_tcp_client.py
-deactivate
 ```
 
-Output should be similar to the following, but columns and predictions will match the data in your scoring pipeline:
+The output should be similar to the following, but columns and predictions will match the data in your scoring pipeline:
 
 ![tcp-client-get-scores](assets/tcp-client-get-scores.jpg)
 
+Once you are done, you can deactivate your virtual environment.
+
+```bash
+deactivate
+```
 
 ## Task 5: Challenge
 
 ### Execute Scoring Pipeline for a New Dataset
 
-There are various challenges one could do, you could do something that helps you in your daily life or job. Maybe there is a dataset you are working with, you could reproduce the steps we did above, but for your dataset, build a new experiment and execute your Python scoring pipeline to do batch scoring or interactive scoring.
+There are various challenges you could do. You could do something that helps you in your daily life or job. Maybe there is a dataset you are working with; you could reproduce the steps we did above, but for your dataset, build a new experiment and execute your Python scoring pipeline to do batch scoring or interactive scoring.
 
 ### Embed Scoring Pipeline into Existing Program
 
-Another challenge could be to use the existing Python scoring pipeline we executed and instead of using the examples we shared above, integrate the scoring pipeline into an existing Java, Python or C++ program using the TCP or HTTP scoring service.
+Another challenge could be to use the existing Python scoring pipeline we executed. Instead of using the examples we shared above, integrate the scoring pipeline into an existing Java, Python, or C++ program using the TCP or HTTP scoring service.
```

Comment:
rewrite to this: "Another challenge could be to use the existing Python scoring pipeline we executed: instead of using the examples we shared above, integrate the scoring pipeline into an existing Java, Python, or C++ program using the TCP or HTTP scoring service."